File: ee-schkola/src-gen/main/kotlin/ee/schkola/SharedApiBase.kt
package ee.schkola

import java.util.Date




open class SchkolaBase {
    val trace: Trace
    val id: String


    constructor(id: String = "") {
        this.id = id
        this.trace = Trace()
    }

    companion object {
        val EMPTY = SchkolaBase()
    }
}


open class Trace {
    var createdAt: Date
    var updatedAt: Date
    var modifiedBy: String


    constructor(createdAt: Date = Date(), updatedAt: Date = Date(), modifiedBy: String = "") {
        this.createdAt = createdAt
        this.updatedAt = updatedAt
        this.modifiedBy = modifiedBy
    }

    companion object {
        val EMPTY = Trace()
    }
}

File: ee-schkola/src-gen/main/go/ee/schkola/SharedApiBase.go
package schkola

import (
    "github.com/looplab/eventhorizon"
    "time"
)

type SchkolaBase struct {
    *Trace
    Id  eventhorizon.UUID
}

func NewSchkolaBase(id eventhorizon.UUID) (ret *SchkolaBase) {
    ret = &SchkolaBase{
        Id : id,
    }
    
    return
}







type Trace struct {
    CreatedAt  *time.Time
    UpdatedAt  *time.Time
    ModifiedBy  string
}

func NewTrace(createdAt *time.Time, updatedAt *time.Time, modifiedBy string) (ret *Trace) {
    ret = &Trace{
        CreatedAt : createdAt,
        UpdatedAt : updatedAt,
        ModifiedBy : modifiedBy,
    }
    
    return
}





